### ISSUE TITLE
darksage code logic

### ISSUE THREAD

**Issue by manodeep** [CAS-eResearch/TAOImport#27]
@arhstevens I am running into errors while converting Millennium at this [line](https://github.com/CAS-eResearch/TAOImport/blob/dev/examples/darksage.py#L3895). Do you mind explaining what the code *should* do?

**Comment by arhstevens** [comment]
This is a safety net I put in place in case any properties were output as NaNs or negative (all properties listed in 'check_fields' should be positive and finite).  In those cases, the properties are reset to zero.

Weird that this would cause an error though.  Don't understand why it would unless the syntax doesn't work the same for a different version of Python.

**Comment by arhstevens** [comment]
Are you trying to convert the 2016 or 2018 version?  This code should work for the 2016 version, but the galaxy fields are different for the 2018 one.

**Comment by manodeep** [comment]
Currently converting the 2016 version. I checked out the corresponding 2016 tagged version of DarkSage and used that parameter file.

**Comment by manodeep** [comment]
@arhstevens Separately, the darksage converter is spending a significant amount of time in the functions that compute certain fields, e.g., `totHI, pseudo-bulge mass` etc. Do you have time to sit down and we can figure out more efficient implementations...

**Comment by arhstevens** [comment]
Sure.  I can be over in ~10 mins?
It's definitely way inefficient.  I initially implemented the simplest thing that fitted into the way the code was already structured, rather than spending time optimizing something that might only need to be run once anyway.   Seems like a wise opportunity to improve things together now though.

**Comment by manodeep** [comment]
Yup - the conversion process is taking far longer for darksage. At the current rate, it will take 42+ days to complete Millennium with darksage (on OzSTAR). For comparison, Millennium + sage was ~ 20 hours
